helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1876,6 +1885,14 @@ public ZNRecord update(ZNRecord currentData) {
 
         InstanceConfig config = new InstanceConfig(currentData);
         config.setInstanceEnabled(enabled);
+        if (!enabled) {
+          if (reason!=null) {
+            config.setInstanceDisabledReason(reason);
+          }
+          if (disabledType != null) {
+            config.setInstanceDisabledType(disabledType);
+          }
```

Comment:
Let's have a default type for it. Then it can support backward compatible.

Reply:
As our offline discussion, the default type will be returned when query and won't be written to ZK.